src/OpenTelemetry.AutoInstrumentation/Instrumentations/NoCode/NoCodeIntegration0.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using OpenTelemetry.AutoInstrumentation.CallTarget;
+
+namespace OpenTelemetry.AutoInstrumentation.Instrumentations.NoCode;
+
+/// <summary>
+/// NoCode calltarget instrumentation
+/// </summary>
+public static class NoCodeIntegration0
```

Comment:
I think this approach shows how no code instrumentation can be implemented, but needing to stub out these NoCodeIntegration classes for different parameter counts will limit the types of methods that can be instrumented.

I wonder if there is a way to add some indirection to the implementation. That is, all NoCode instrumentation will have the exact same injected code, instead of injecting different code to call methods with different numbers of parameters. If we can do that, I think the implementation can be simplified. If we do not need access to the method parameters, I think this indirect approach can be simple. If we do need access to the parameters we may need to leverage arrays with boxing or Tuples with boxing in order to provide access to the parameters when necessary.